templates/telemetry_derived/clients_scalar_aggregates_v1/init.sql

```diff
@@ -0,0 +1,14 @@
+CREATE TABLE IF NOT EXISTS
+  `moz-fx-data-shared-prod.telemetry_derived.clients_scalar_aggregates_v1` (
+    client_id STRING,
+    os STRING,
+    app_version INT64,
+    app_build_id STRING,
+    channel STRING,
+    scalar_aggregates ARRAY <STRUCT<metric STRING,
+    metric_type STRING,
+    key STRING,
+    agg_type STRING,
+    value FLOAT64>>)
+PARTITION BY RANGE_BUCKET(app_version, GENERATE_ARRAY(30, 200, 1))
+CLUSTER BY app_version, channel, client_id
```

Comment:
If app_version is a discrete integer used for partitioning already, I don't think there's any benefit from including it as a clustering field. There may be some small performance penalty for including it.
```suggestion
CLUSTER BY channel, client_id
```

Reply:
Yes, unfortunately this is a bit strange, but I'll explain why I ended up doing this.

When I created a partition by `app_version`  and require that queries filter by partition, if I do something like `WHERE app_version >= (latest_version - 2)` (which is how the filtering is done in the query now) it complains that I'm not filtering by app version

It's only happy if I explicitly say `WHERE app_version > 50` for example. Not surprising I guess since this feature is in beta. This made me suspect that we're not getting partitioning gains by partitioning on `app_version` and that maybe we can make up for it by clustering on it as well. I haven't confirmed this theory. 

However, in response to your question above, I've confirmed performance gains overall when incorporating all the changes in this PR.

Reply:
We'll only ever have three app_version values in the table at a time, correct? If that's true, I don't see much benefit from partitioning on app_version. It's best to avoid a beta feature if possible, too. And it looks like we'd be able to avoid all the casting between int and string if we don't use app_version as a partitioning field?

Clustering requires partitioning, but we could introduce a dummy date-type column to be the partitioning column if needed.

> When I created a partition by app_version and require that queries filter by partition, if I do something like WHERE app_version >= (latest_version - 2) (which is how the filtering is done in the query now) it complains that I'm not filtering by app version

This may be a limitation for date partitioning too, if the value you're trying to filter on can't be known until looking up results from a subquery.

Reply:
It can have up to 5 values for `app_version` at a time since the `latest_version` is different depending on the channel. Even if we don't cluster on `app_version`, it would still be useful to be able to cluster on the other fields.

I like the dummy date-type column idea. I will file a follow-up issue to switch to that. 
Would just like to get the ball rolling on this to see how it goes, so I'll circle back to fix this.